toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -449,6 +458,18 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		}
 	}
 
+	// Add groups
+	err = addGroups(installChroot, config.Groups)
```

Comment:
It looks like this PR is making 2 sets of changes at once:

1) Allowing specification of custom home dir + primary group for user accounts created.
2) Shift user/group creation *earlier* in the image build process, such that it happens right after installing `filesystem` (and now `shadow-utils`, which is required for `useradd`).

From a PR perspective, it's probably cleanest to split these into separate commits/PRs. (That also would make it easier to identify problems down the road / revert logic if one half of the changes becomes an issue.)

From a code change perspective, the second change described above seems more impactful than the first. Can we think of what unintended behavioral deltas could result? For example:

* With the explicit installation of `shadow-utils`, is there a risk that someone who had pinned `shadow-utils` to a specific package version in their `.json` package list file now may end up with odd behavior during image build? 
* I see the previous logic didn't always invoke `addGroups()`. I'm not sure I understand why at first glance, but with this change, we'd now always call it -- is that okay?

Reply:
Makes sense, thanks for the suggestion!
I've separated those and posted the PR for the second part here: https://github.com/microsoft/CBL-Mariner/pull/6957